cdisc_rules_engine/services/data_services/local_data_service.py

```diff
@@ -52,7 +54,11 @@ def has_all_files(self, prefix: str, file_names: List[str]) -> bool:
 
     @cached_dataset(DatasetTypes.CONTENTS.value)
     def get_dataset(self, dataset_name: str, **params) -> pandas.DataFrame:
-        reader = self._reader_factory.get_service()
+        # print("dataset_name = ", dataset_name)
+        if dataset_name.endswith(".json"):
+            reader = self._reader_factory.get_service("json")
+        else:
+            reader = self._reader_factory.get_service()
```

Comment:
The idea of the factory is to be able to identify the right service based on the file extension. This if should be in the factory, not in the data service

Reply:
Dear @alexfurmenkov, please feel free to change this in the code yourself. I am an absolute beginner in Python, and do not know how this should be done in the factory. Also, I do not have time to dive into this in the next two weeks due to some family urgencies. Many thanks in advance!!!

Reply:
@JozefAerts please ping me as you are back and I will help you to address this comment.

Reply:
Ping... I am back ... If you can help me here, yes, please. But also feel free to improve the code yourself, so that I can learn from what you did.

Reply:
@JozefAerts are you familiar with factory pattern?

Reply:
No, unfortunately I am not, and surely not in Python. There is something similar in Java, but also there my knowledge is close to zero ...

Reply:
Dear @alexfurmenkov,
To make progress (as this will be be important for acceptance of the modern Dataset-JSON format by FDA), I propose that the current code is accepted to flow in "development", and that we do the optimization with the factory in the next round.
Alternatively, if someone (like you) adds the "factory" for the Dataset-JSON case, I can use that to further optimize the code so that we don't need any "if" statements at all for checking the file suffix.
Many thanks for all your super work!

Reply:
I don't think we should just pull this in. We are in no rush to get dataset json functionality released. Within the next couple weeks I can spend some time cleaning this up. Potentially on a new branch

Reply:
Thanks! I will check with Sam and Nick DD in whether this should get any priority.
If you can clean up in the next weeks, that would indeed be great!

Reply:
@nhaydel @JozefAerts thanks guys, please let me know as the PR in the new branch is ready